src/controller.coffee

```diff
@@ -817,26 +820,50 @@ Editor::redo = ->
   currentValue = @getSerializedEditorState()
 
   until @redoStack.length is 0 or
-      (@redoStack[@redoStack.length - 1] is 'CAPTURE' and
+      (@redoStack[@redoStack.length - 1] instanceof CapturePoint and
       not @getSerializedEditorState().equals(currentValue))
     operation = @popRedo()
     if operation instanceof FloatingOperation
       @performFloatingOperation(operation, 'forward')
     else
       @getDocument(operation.document).perform(
-        operation.operation, 'forward', (if operation.document is @cursor.document then [@cursor.location] else [])
-      ) unless operation is 'CAPTURE'
+        operation.operation, 'forward', @getPreserves(operation.document)
+      ) unless operation instanceof CapturePoint
+
+  if @undoStack[@undoStack.length - 1] instanceof CapturePoint
+    @rememberedSockets = @undoStack[@undoStack.length - 1].rememberedSockets.map (x) -> x.clone()
 
   @popRedo()
   @redrawMain()
   return
 
 Editor::undoCapture = ->
-  @pushUndo 'CAPTURE'
+  @pushUndo new CapturePoint(@rememberedSockets)
+
+class CapturePoint
```

Comment:
I'll add some comments. This was to preserve the remembered socket contents across undo and redo.